configure

```diff
@@ -581,104 +1093,104 @@ return __builtin_popcountl(argc) == 100;
-      printf "%s" "$MSG" >> config.log
-      touch build/test-popcnt.c
-      rm build/test-popcnt
-      $CC -mpopcnt build/test-popcnt.c -o ./build/test-popcnt > /dev/null 2>&1
-      build/test-popcnt > /dev/null 2>&1
-      if [ $? -eq 0 ]; then
-         printf "%s\n" "yes"
-         echo "yes" >> config.log
-	 POPCNT_FLAG="-mpopcnt"
-      else
-         printf "%s\n" "no"
-	 echo "no" >> config.log
-      fi
-      rm -f build/test-popcnt{,.c}
-   #in case -mpopcnt is not available, the test program will use an illegal
-   #instruction and that will print out something on stderr when the if
-   #construction is exited, whence the following "2> /dev/null"
-   fi 2> /dev/null
+    printf "%s\n" "yes"
+    echo "yes" >> config.log
+    CONFIG_POPCNT_INTRINSICS="#define FLINT_USES_POPCNT"
+
+    if [ "$BUILD_MACHINE" = "x86_64" ]; then
+        MSG="Testing native popcount..."
+        printf "%s" "$MSG"
+        printf "%s" "$MSG" >> config.log
+        touch build/test-popcnt.c
+        rm build/test-popcnt
+        $CC -mpopcnt build/test-popcnt.c -o ./build/test-popcnt > /dev/null 2>&1
+        build/test-popcnt > /dev/null 2>&1
+        if [ $? -eq 0 ]; then
+            printf "%s\n" "yes"
+            echo "yes" >> config.log
+            POPCNT_FLAG="-mpopcnt"
+        else
+            printf "%s\n" "no"
+            echo "no" >> config.log
+        fi
+        rm -f build/test-popcnt{,.c}
+        #in case -mpopcnt is not available, the test program will use an illegal
+        #instruction and that will print out something on stderr when the if
+        #construction is exited, whence the following "2> /dev/null"
+    fi 2> /dev/null
 else
-   rm -f build/test-popcnt.c
-   printf "%s\n" "no"
-   echo "no" >> config.log
+    rm -f build/test-popcnt.c
+    printf "%s\n" "no"
+    echo "no" >> config.log
 fi
 
 #defaults for CFLAGS
 if [ -z "$CFLAGS" ]; then
-   if [ "$OS" = "MINGW64" ]; then
-      CFLAGS="-std=c99 -O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-      ANSI_FLAG=""
-   elif [ "$OS" = "CYGWIN64" ]; then
-      CFLAGS="-O2 -funroll-loops -g -D _WIN64 $POPCNT_FLAG $ABI_FLAG"
-      ANSI_FLAG=""
-   elif [ "$MACHINE" = "mips64" ]; then
-      CFLAGS="-O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-      ANSI_FLAG=""
-   elif test "$KERNEL" = "FreeBSD" -o "$OS" = "OpenBSD"; then
-      CFLAGS="-std=c99 -pedantic -Wall -O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-      ANSI_FLAG=""
-      CXXFLAGS="-std=c++11 -pedantic -Wall -O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-   else
-      ANSI_FLAG="-ansi"
-      CFLAGS="-pedantic -Wall -O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-   fi
+    if [ "$BUILD_OS" = "MINGW64" ]; then
+        CFLAGS="-std=c99 -O2 $POPCNT_FLAG $ABI_FLAG $DEBUGGING_FLAG $MARCH_FLAG"
+        ANSI_FLAG=""
+    elif [ "$BUILD_OS" = "CYGWIN64" ]; then
+        CFLAGS="-O2 -D _WIN64 $POPCNT_FLAG $ABI_FLAG $DEBUGGING_FLAG $MARCH_FLAG"
+        ANSI_FLAG=""
+    elif [ "$BUILD_MACHINE" = "mips64" ]; then
```

Comment:
It's really weird that this `if/elif` construct mixes checks for the OS with a check for the "machine" ("CPU"?) type. Makes me wonder what will happen when someone runs FreeBSD on a MIPS64 (not that I think it likely that any of the 3 folks doing that will want to use FLINT :joy:)

Reply:
Lol, I agree.